gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/api/ConfigStore.java

```diff
@@ -32,30 +48,54 @@
   public String getCurrentVersion();
 
   /**
-   * @return the configuration store root URI
+   * Obtains the config store root URI. This typically represents the physical location of the store.
+   * It may contain the server name and the path to the physical location of the store.
+   * @return the configuration store root URI .
    */
   public URI getStoreURI();
 
   /**
-   * @param uri - the uri relative to this configuration store
-   * @param version - specify the configuration version in the configuration store.
-   * @return - the direct children URIs for input uri against input configuration version
+   * Obtains the direct children config keys for a given config key. For example, the child
+   * paths for /data/tracking may be /data/tracking/ImpressionEvent and /data/tracking/ClickEvent .
+   *
+   * <p>Note that this method should not be used for "service discovery", i.e. it need not return
+   * all possible child paths but only those defined in the store. For example, the configuration
+   * for /data/tracking/ConversionEvent may be implicitly inherited from /data/tracking and
+   * /data/tracking/ConversionEvent may not be returned by this method.
+   *
+   * @param  configKey      the config key path whose children are necessary.
+   * @param  version        specify the configuration version in the configuration store.
+   * @return the direct children config key paths
+   * @throws VersionDoesNotExistException if the requested config version does not exist (any longer)
    */
-  public Collection<URI> getChildren(URI uri, String version) throws VersionDoesNotExistException;
+  public Collection<ConfigKeyPath> getChildren(ConfigKeyPath configKey, String version)
+      throws VersionDoesNotExistException;
 
   /**
-   * @param uri - the uri relative to this configuration store
-   * @param version - specify the configuration version in the configuration store.
-   * @return - the directly imported URIs for input uri against input configuration version
-   *  the earlier URI in the List will have higher priority when resolving configuration conflict
+   * Obtains the list of all config keys with which are given config key is tagged/annotated.
+   *
+   * @param  configKey      the config key path whose tags are needed
+   * @param  version        the configuration version in the configuration store.
+   * @return the paths of the directly imported config keys for the specified config key and
+   * version. Note that order is significant the earlier URI in the List will have higher priority
+   * when resolving configuration conflicts.
+   * @throws VersionDoesNotExistException if the requested config version does not exist (any longer)
+   *
    */
-  public List<URI> getOwnImports(URI uri, String version) throws VersionDoesNotExistException;
+  public List<ConfigKeyPath> getOwnImports(ConfigKeyPath configKey, String version)
```

Comment:
As noted in the javadoc, the order is actually significant since it governs to which configuration to fallback if it is not in the own/immediate configuration object.